pkg/routine/routine_test.go

```diff
@@ -33,13 +32,13 @@ func TestRun_createsNewContext(t *testing.T) {
 }
 
 func TestRun_transfersLogger(t *testing.T) {
-	buf := bytes.Buffer{}
-	logger := log.Output(&buf)
-	ctx := logger.WithContext(context.Background())
+	var sink log.Sink
+	logger := log.Logger()
+	ctx := log.ContextWithSink(logger.WithContext(context.Background()), &sink)
 	waitForRun(ctx, func(ctx context.Context) {
 		log.Ctx(ctx).Debug().Msg("foobar")
 	})
-	require.Contains(t, buf.String(), "foobar")
+	require.Contains(t, string(sink.ToJSON()), "foobar")
 }
```

Comment:
Please don't re-purpose this test. This test did test that the logger is transferred. Add an additional test to test that the sink is transferred, please. The sink cannot go without the logger, but the logger can go without the sink.

Reply:
done